src/OrleansRuntime/Catalog/Catalog.cs

```diff
@@ -1349,6 +1142,11 @@ private void RerouteAllQueuedMessages(ActivationData activation, ActivationAddre
             {
                 RequestContext.Import(requestContextData);
                 await activation.Lifecycle.OnStart();
+                // TODO: Consider pre and post activation stages in lifecycle for PreActivateProtocolParticipant and PostActivateProtocolParticipant like behaviors
+                if (activation.GrainInstance is ILogConsistencyProtocolParticipant)
+                {
+                    await ((ILogConsistencyProtocolParticipant)activation.GrainInstance).PreActivateProtocolParticipant();
```

Comment:
question: shouldn't this happen before calling `activation.Lifecycle.OnStart();`

Reply:
The expected behavior is that state be loaded then pre-activate, activate, post-activate.  Since state load happens in lifecycle, this will need to be after onStart, for now.  When onActivateAsync is called by the lifecycle, this can be as well.